### PR TITLE
Simple Port Management

### PR DESCRIPTION
**Pull request by mysticall**
Simple Port Management patches will allow you to modify the WAN port.
This functionality is good for ISPs with IPTV.
In most cases, it's not necessary to install a switch at the customer.

### CODE REVIEW

**Comment by lantis1008** [comment]
Hi,
I'm not sure that these patches suit the larger gargoyle audience. They look to be very specific to a few routers, and a few scenarios.

My thoughts:
- worried about your removal of the issinglesthernet checks in basic.js. The Gargoyle Pocket Router (when it was in the shop) is a single Ethernet device. Breaking functionality on a device sold by Gargoyle is not good. 
- your handling of VLAN switch config is not generic. It only applies to those particular devices
- multiuser support, while I'm not disinclined to this, the way this one is implemented leaving a large security backdoor is a big no from me
- removal of firstboot page means users aren't promoted to set language or change default password
- I don't like the idea of having patches to the main gargoyle package. That seems hard to maintain

To me this seems like a very niche patch that would be better implemented in a forked repo by creating a new target and using the target/patches/ structure to apply these at build time

File: spm/001-SPM-generic.diff [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ 
+diff --git a/package/gargoyle/files/www/js/basic.js b/package/gargoyle/files/www/js/basic.js
+index cce47d4a..18a3cae4 100644
+--- a/package/gargoyle/files/www/js/basic.js
++++ b/package/gargoyle/files/www/js/basic.js
+@@ -163,10 +163,6 @@ function saveChanges()
+ 					uci.remove('network', 'wan', 'ifname');
+ 					uci.set('network', 'wan', 'type', 'bridge');
+ 				}
+-				else if( singleEthernetIsWan() )
```
Comment (lantis1008) [inline spm/001-SPM-generic.diff]: Broken singleethernetwan devices

File: spm/010-SPM-ramips-gargoyle-WE526-reverse.diff [hunk 2]
```diff
@@ -0,0 +1,55 @@
+diff --git a/package/gargoyle/files/www/basic.sh b/package/gargoyle/files/www/basic.sh
+index 0a69453c..75f613ed 100755
+--- a/package/gargoyle/files/www/basic.sh
++++ b/package/gargoyle/files/www/basic.sh
+@@ -390,6 +400,18 @@ var isb43 = wirelessDriver == "mac80211" && (!GwifiN) ? true : false ;
+ 			</div>
+ 
+ 			<div class="panel-body">
++				<div class="row form-group" >
++					<label for="wan_protocol" class="col-xs-5" >WAN ports:</label>
```
Comment (lantis1008) [inline spm/010-SPM-ramips-gargoyle-WE526-reverse.diff]: Inconsistent use of i18n

File: spm/010-SPM-ramips-gargoyle-WE526-reverse.diff [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+diff --git a/package/gargoyle/files/www/js/basic.js b/package/gargoyle/files/www/js/basic.js
+index cce47d4a..18a3cae4 100644
+--- a/package/gargoyle/files/www/js/basic.js
++++ b/package/gargoyle/files/www/js/basic.js
+@@ -189,7 +185,26 @@ function saveChanges()
+ 				}
+ 				else
+ 				{
+-					uci.set('network', 'wan', 'ifname', defaultWanIf);
++					if(getSelectedValue("wan_ports").match(/WLLLL/) && (VLAN1_PORTS != "1 2 3 4 6t" || VLAN2_PORTS != "0 6t"))
```
Comment (lantis1008) [inline spm/010-SPM-ramips-gargoyle-WE526-reverse.diff]: VLAN config only suits some devices

File: spm/030-password-to-text.diff [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+--- a/package/gargoyle/files/www/basic.sh
++++ b/package/gargoyle/files/www/basic.sh
+@@ -351,9 +363,7 @@ var isb43 = wirelessDriver == "mac80211" && (!GwifiN) ? true : false ;
+ 				<div id="bridge_pass_container" class="row form-group">
+ 					<label class="col-xs-5" for="bridge_pass" id="bridge_pass_label"><%~ Pswd %>:</label>
+ 					<span class="col-xs-7">
+-						<input type="password" id="bridge_pass" class="form-control" size="20" onkeyup="proofreadLengthRange(this,8,999)"/>
+-						<input type="checkbox" id="show_bridge_pass" onclick="togglePass('bridge_pass')" />
+-						<label for="show_bridge_pass" id="show_bridge_pass_label"><%~ rvel %></label>
++						<input type="text" id="bridge_pass" class="form-control" size="20" onkeyup="proofreadLengthRange(this,8,999)"/>
```
Comment (lantis1008) [inline spm/030-password-to-text.diff]: I don't like this change

File: spm/031-disable-firstboot-page.diff [hunk 5]
```diff
@@ -0,0 +1,15 @@
+diff --git a/package/gargoyle/files/etc/config/gargoyle b/package/gargoyle/files/etc/config/gargoyle
+index 2be0104..27bb84c 100644
+--- a/package/gargoyle/files/etc/config/gargoyle
++++ b/package/gargoyle/files/etc/config/gargoyle
+@@ -7,8 +7,9 @@ config global global
+ 	option common_css	'common.css'
+ 	option common_js	'common.js'
+ 	option require_web_password	'1'
+-	option is_first_boot		'1'
```
Comment (lantis1008) [inline spm/031-disable-firstboot-page.diff]: Removal of firstboot and hardcodd time date

File: spm/041-MLS-passwd.diff [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
++++ b/patches-generic/005-passwd.patch
+@@ -1,9 +1,17 @@
+ --- a/package/base-files/files/etc/shadow
+ +++ b/package/base-files/files/etc/shadow
+-@@ -1,5 +1,5 @@
++@@ -1,4 +1,5 @@
+ -root::0:0:99999:7:::
+-+root:$1$V4UetPzk$CYXluq4wUazHjmCDBCqXF.:0:0:99999:7:::
+++root:$1$PsM6bycH$esmgIpPUjPhSYCtCASne91:0:0:99999:7:::
+++admin:$1$PsM6bycH$esmgIpPUjPhSYCtCASne91:0:0:99999:7:::
```
Comment (lantis1008) [inline spm/041-MLS-passwd.diff]: Weak default passwords on TWO users with no forced change

File: spm/309-default-settings.diff [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+ 
+-config wifi-iface
++config wifi-iface ap_g
+ 	option device   radio$devidx
+ 	option network  lan
+ 	option mode     ap
+-	option ssid     OpenWrt
++	option disassoc_low_ack 0
++	option wds 1
++	option ssid     Gargoyle-ISP
```
Comment (lantis1008) [inline spm/309-default-settings.diff]: Very specific setup here not applicable to all users

File: spm/309-default-settings.diff [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+index 7be01d25b8..e866130b07 100644
+--- a/package/network/config/firewall/files/firewall.config
++++ b/package/network/config/firewall/files/firewall.config
+@@ -172,6 +172,18 @@ config rule
+ 	option proto		udp
+ 	option target		ACCEPT
+ 
++config remote_accept 'ra_443_443'
++	option local_port '443'
++	option remote_port '443'
```
Comment (lantis1008) [inline spm/309-default-settings.diff]: Default holes in firewall

**Comment by mysticall** [comment]
`    I don't like the idea of having patches to the main gargoyle package. That seems hard to maintain

To me this seems like a very niche patch that would be better implemented in a forked repo by creating a new target and using the target/patches/ structure to apply these at build time`

I can periodically update the patches in my forked repository.

These patches provide advanced functionality of the standard system. Any functionality or change is exported to a **separate patch and described**. They are intended for use with custom build. The user can decide which patch wants to use and who does not.
The patches offer users with no experience an easy way to change basic things without losing their own time with a thorough examination of the system.

**Comment by ericpaulbishop** [comment]
I am going to close this pull request, as I'm not willing to make such radical changes to the base system for such a niche application.

If you can figure out a way to do this as a plugin, I will reconsider merging it.

I would add that if you find that you can't do it as a plugin due to changes necessary in the basic section (which may be the case), if you can find a way to make it possible to preserve current functionality by default, but allow a plugin to modify certain sections of the basic page I would consider merging those changes to connection/basic. I know, that's not easy/straightforward, but if you want to go that route you might start with looking at how plugins can add sections to the login/landing page right now and do something similar in connection/basic.

**Pull request closed without merging**